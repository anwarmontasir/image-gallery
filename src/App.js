import React, { Component } from 'react';
import './App.css';
import { bunnyBuild } from './services/bunnies';
import { List } from './Components/List';
import { Thumbs } from './Components/Thumbs';
import { Gallery } from './Components/Gallery';

function listView({ bunnies }) {
  return <div><h2>List View</h2> <List listArray={ bunnies } /> </div>;
}
function thumbView({ bunnies }) {
  return <div><h2>Thumbnail View</h2> <Thumbs thumbArray={ bunnies } /> </div>;
}
function galleryView({ bunnies }) {
  return <div><h2>Gallery View</h2> <Gallery galleryArray={ bunnies } /> </div>;
}

const viewDict = {
  list: listView,
  thumbs: thumbView,
  gallery: galleryView
}

const viewArray = Object.keys(viewDict);

class App extends Component {

  constructor() {
    super();
    this.state = {
      bunnies: bunnyBuild(),
      view: viewArray[0],
      views: viewArray
    };
  }

  render() {
    const { bunnies, views, view } = this.state;
    const ViewWrapper = viewDict[view];

    return (
      <main>
        <header>
          <h1>Cute Bunnies App</h1>
        </header>
        <section>
          {views.map(v => (
            <button key={v} onClick={() => this.setState({ view: v })}>
              {v}
            </button>
          ))}
        </section>
        <section>
          <ViewWrapper bunnies={ bunnies } />
        </section>
      </main>
    );
  }
}

export default App;
