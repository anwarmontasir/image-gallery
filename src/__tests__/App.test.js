import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Routes } from '../App';
import { MemoryRouter } from 'react-router-dom';

import App from '../App';

describe('App works', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });
  
  it('renders a snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Routes work', () => {
  it('Defaults to home', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Routes />
      </MemoryRouter>
    );
    const home = wrapper.find('Home');
    expect(home.length).toBe(1);
  });

  it('Goes to about', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/about']}>
        <Routes/>
      </MemoryRouter>
    );
    const home = wrapper.find('About');
    expect(home.length).toBe(1);
  });
});
