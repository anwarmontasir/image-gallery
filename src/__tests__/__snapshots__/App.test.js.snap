// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App works renders a snapshot 1`] = `
<div>
  <header>
    <h1>
      Cute Bunnies: The App
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/images?view=list"
            onClick={[Function]}
          >
            Images
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <div>
      <h1>
        Bunny App Home Placeholder
      </h1>
    </div>
  </main>
</div>
`;
